spec/components/devolved_nations_spec.rb

```diff
@@ -0,0 +1,75 @@
+require "rails_helper"
+
+describe "Devolved Nations", type: :view do
+  def component_name
+    "devolved_nations"
+  end
+
+  it "renders a devolved nations component, which applies to one nation, correctly" do
+    render_component(
+      national_applicability: {
+        england: {
+          applicable: true,
+        },
+      },
+    )
+    assert_select ".gem-c-devolved-nations > h2", text: "Applies to England"
+  end
+
+  it "renders a devolved nations component, which applies to two nations, correctly" do
+    render_component(
+      national_applicability: {
+        england: {
+          applicable: true,
+        },
+        wales: {
+          applicable: true,
+        },
+      },
+    )
+    assert_select ".gem-c-devolved-nations > h2", text: "Applies to England and Wales"
+  end
+
+  it "renders a devolved nations component, which applies to three nations, correctly" do
+    render_component(
+      national_applicability: {
+        england: {
+          applicable: true,
+        },
+        scotland: {
+          applicable: true,
+        },
+        wales: {
+          applicable: true,
+        },
+      },
+    )
+    assert_select ".gem-c-devolved-nations > h2", text: "Applies to England, Scotland and Wales"
+  end
+
+  it "renders a devolved nations component, which applies to one nation, with individual guidance available, correctly" do
+    render_component(
+      national_applicability: {
+        england: {
+          applicable: true,
```

Comment:
Just out of curiousity, what happens if you pass two applicable `true` nations and one `false`? Should there be a test for that?

Reply:
This would be the same as the following as the scenario below, since nations are only filtered on `applicable: true` values, and `false` values are ignored, so _I think_ we are probably get by without the additional test.

```
<%= render "govuk_publishing_components/components/devolved_nations", {
  national_applicability: {
    england: {
      applicable: true
    },
    wales: {
      applicable: true
    }
  }
} %>
```

Reply:
Sorry, I meant if you pass two `true` countries and one `false` but also with an `alternative_url`. I've just checked, and it's possible to do this:

![Screenshot 2021-09-07 at 14 30 15](https://user-images.githubusercontent.com/861310/132353267-79f1fde0-46be-4cff-8e3b-acbe2d748dbd.png)

I don't know if this should be possible? It seems okay (it's basically the `Applies to one nation individual guidance available` but with more than one true nation). If this is meant to be possible, then I think it needs a test.

Reply:
I've found a similar example here: https://www.gov.uk/government/publications/record-your-private-practice-when-youre-learning-to-drive (3 nations and 1 guidance) so i've used this one for an additional example (and associated tests in `devolved_nations_helper_spec.rb` and `devolved_nations_spec.rb`).